sw/device/lib/crypto/impl/hash.c

```diff
@@ -18,15 +18,50 @@
 /**
  * Ensure that the hash context is large enough for the SHA-512 state struct.
  */
+static_assert(sizeof(hash_context_t) >= sizeof(sha384_state_t),
+              "hash_context_t must be big enough to hold sha384_state_t");
 static_assert(sizeof(hash_context_t) >= sizeof(sha512_state_t),
               "hash_context_t must be big enough to hold sha512_state_t");
 /**
  * Ensure that the SHA-512 state struct is suitable for `hardened_memcpy()`.
  */
+static_assert(sizeof(sha384_state_t) % sizeof(uint32_t) == 0,
+              "Size of sha384_state_t must be a multiple of the word size for "
+              "`hardened_memcpy()`");
 static_assert(sizeof(sha512_state_t) % sizeof(uint32_t) == 0,
               "Size of sha512_state_t must be a multiple of the word size for "
               "`hardened_memcpy()`");
 
+/**
+ * Save a SHA-384 state to a generic hash context.
+ *
+ * @param[out] ctx Generic hash context to copy to.
+ * @param state SHA-384 context object.
+ */
+static void sha384_state_save(hash_context_t *const restrict ctx,
```

Comment:
Should this have the const swapped in the function parameters given how we are calling `hardened_memcpy`? 

```c
// cxt->data <= state
sha384_state_save(
  hash_context_t *restrict ctx,
  const sha384_state_t *restrict state)
```

Reply:
I'll swap this in the PR that adds SHA-256 to `hash.c` just to avoid waiting for the CI again; I already have that PR ready locally and it touches the same code.